src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1282,8 +1281,8 @@ bool isRefEscape
                         break;
                     }
 
-                    var refKind = argRefKindsOpt.IsDefault ? RefKind.None : argRefKindsOpt[argIndex];
-                    if (refKind != RefKind.None && argument.Type?.IsByRefLikeType == true)
+                    var paramIndex = argsToParamsOpt.IsDefault ? argIndex : argsToParamsOpt[argIndex];
+                    if (parameters[paramIndex].RefKind != RefKind.None && argument.Type?.IsByRefLikeType == true)
```

Comment:
Check `GetInvocationEscapeScope()` and `GetEffectiveRefKindAndMarkMatchedInParameter()`